### ISSUE TITLE
Ability to disable pluralization and decamelize

### ISSUE THREAD

**Issue by adamlc** [wopian/kitsu#69]
Thanks for a great library! 

It would be awesome if we could disable pluralization and decamelize when requesting resources. I'm using the library on my own API, but unfortunately one of those functions is breaking my ids (MongoID) in the request.

**Comment by wopian** [comment]
This is definitely a feature that's needed. The abandoned 3.2.0 release candidates removed pluralisation as a temporary fix for a small project I did (but those releases are full of other breaking issues).

This should hopefully be part of the upcoming 3.2.0 release with the ability to pass `pluralize` extra rules by exposing its API.

**Comment by wopian** [comment]
Started working on this as part of `4.0.0` to further clean up the code and reduce bundle bloat. 👍

**Comment by wopian** [comment]
Now implemented for the method endpoints. Just 1 internal usage that needs resolving and is somewhat blocked by #81.

`4.0.0` release should be out some point this week

**Comment by wopian** [comment]
@adamlc whats your thoughts on the endpoints and `type` handling using the same option? 

- `camelcase: true`
	1, endpoints are decamelized (`libraryEntries` becomes `library-entries`)
	2. guessed `type` value is camelcased for `POST`, `PATCH` and `DELETE` (`library-entries` become `libraryEntries`)
- `camelcase: false`
	1. endpoints are kept as-is
	2. guessed type is kept as-is

Note: `camelcase` option is currently `decamelize` in source

**Comment by wopian** [comment]
Implemented in the `4.0.0` release:

```js
new Kitsu({
  pluralize: false,
  camelCaseTypes: false, 
  resourceCase: 'none' // ('kebab' for kebab-case or 'snake' for snake_case) 
})
```
